docs/_docs/reference/contextual/derivation-macro.md

```diff
@@ -21,185 +17,175 @@ trait Eq[T]:
   def eqv(x: T, y: T): Boolean
 ```
 
-we need to implement a method `Eq.derived` on the companion object of `Eq` that
-produces a quoted instance for `Eq[T]`. Here is a possible signature,
+We need to implement an inline method `Eq.derived` on the companion object of `Eq` that calls into a macro to produce a quoted instance for `Eq[T]`.
+Here is a possible signature:
+
 
 ```scala
-given derived[T: Type](using Quotes): Expr[Eq[T]]
+inline def derived[T]: Eq[T] = ${ derivedMacro[T] }
+
+def derivedMacro[T: Type](using Quotes): Expr[Eq[T]] = ???
 ```
 
-and for comparison reasons we give the same signature we had with `inline`:
+Note, that since a type is used in a subsequent macro compilation stage it will need to be lifted to a `quoted.Type` by using the corresponding context bound (seen in `derivedMacro`).
+
 
+For comparison, here is the signature of the inline `derived` method from the [main derivation page](./derivation.md):
 ```scala
-inline given derived[T](using Mirror.Of[T]): Eq[T] = ???
+inline def derived[T](using m: Mirror.Of[T]): Eq[T] = ???
 ```
 
-Note, that since a type is used in a subsequent stage it will need to be lifted
-to a `Type` by using the corresponding context bound. Also, note that we can
-summon the quoted `Mirror` inside the body of the `derived` thus we can omit it
-from the signature. The body of the `derived` method is shown below:
+Note that the macro-based `derived` signature does not have a `Mirror` parameter.
+This is because we can summon the `Mirror` inside the body of `derivedMacro` thus we can omit it from the signature.
+The body of the `derivedMacro` method is shown below:
 
 
 ```scala
-given derived[T: Type](using Quotes): Expr[Eq[T]] =
+def derivedMacro[T: Type](using Quotes): Expr[Eq[T]] =
   import quotes.reflect.*
 
   val ev: Expr[Mirror.Of[T]] = Expr.summon[Mirror.Of[T]].get
 
   ev match
     case '{ $m: Mirror.ProductOf[T] { type MirroredElemTypes = elementTypes }} =>
-      val elemInstances = summonAll[elementTypes]
+      val elemInstances = summonInstances[T, elementTypes]
       def eqProductBody(x: Expr[Product], y: Expr[Product])(using Quotes): Expr[Boolean] = {
         elemInstances.zipWithIndex.foldLeft(Expr(true)) {
           case (acc, ('{ $elem: Eq[t] }, index)) =>
             val indexExpr = Expr(index)
             val e1 = '{ $x.productElement($indexExpr).asInstanceOf[t] }
             val e2 = '{ $y.productElement($indexExpr).asInstanceOf[t] }
             '{ $acc && $elem.eqv($e1, $e2) }
-         }
+        }
       }
-      '{ eqProduct((x: T, y: T) => ${eqProductBody('x.asExprOf[Product], 'y.asExprOf[Product])}) }
+      '{ eqProduct((x, y) => ${eqProductBody('x.asExprOf[Product], 'y.asExprOf[Product])}) }
```

Comment:
Why remove explicit type annotations here? They help understanding.

Reply:
sure I can add that back, it just seemed that typical scala code does not use explicit types on lambdas

Reply:
There's typical Scala code, and there's documentation code. When you're reading documentation, there's no IDE to tell you the types of things. We have to be more explicit. The reader shouldn't have to play type inference.